### PR TITLE
WIP: Job download data file

### PR DESCRIPTION
**Pull request by e-carlin**

### CODE REVIEW

File: sirepo/job.py [hunk 1]
```diff
@@ -32,7 +32,8 @@
 #: path supervisor registers to receive requests from server
 SERVER_URI = '/server'
 
-AGENT_URI = '/agent'
```
Comment (robnagler) [inline sirepo/job.py]: I think  this line shouldn't be deleted

File: sirepo/runner_api.py [hunk 2]
```diff
@@ -23,6 +24,26 @@
...
+        type=simulation_type,
+    )
+    f, c, t = sirepo.template.import_module(sim.type).get_data_file(
+        simulation_db.simulation_run_dir(sim.req_data),
+        sim.sim_data.compute_model(sim.model),
+        int(frame),
+        options=sim.req_data.copy().update(suffix=suffix),
+    )
+    return http_reply.headers_for_no_cache(
+        http_reply.as_attachment(flask.make_response(c), t, f),
```
Comment (robnagler) [inline sirepo/runner_api.py]: This should be app.response_class, not flask.make_response. Encapsulate this complete in http_reply.

File: sirepo/job_api.py [hunk 3]
```diff
@@ -29,6 +31,21 @@
...
 
+@api_perm.require_user
+def api_downloadDataFile(simulation_type, simulation_id, model, frame, suffix=None):
+#TODO(robnagler) validate suffix and frame
+    sim = sirepo.http_request.parse_params(
+        id=simulation_id,
+        model=model,
+        type=simulation_type,
+    )
+    f = _request(data=PKDict(sim.req_data, frame=frame, computeJobHash='x')).file
```
Comment (robnagler) [inline sirepo/job_api.py]: Need to check the response to be "ok", and only then, read file. Sometimes data files aren't there.

File: sirepo/job_api.py [hunk 4]
```diff
@@ -89,4 +108,5 @@ def _request_data(kwargs):
         libDir=lambda: str(sirepo.simulation_db.simulation_lib_dir(b.simulationType)),
         mpiCores=lambda: sirepo.mpi.cfg.cores if b.isParallel else 1,
         userDir=lambda: str(sirepo.simulation_db.user_dir_name(b.uid)),
+        tmpDir=lambda: str(sirepo.simulation_db.tmp_dir())
```
Comment (robnagler) [inline sirepo/job_api.py]: Don't always add tmp_dir see above.

File: sirepo/job_api.py (hunk 3 above)
Comment (robnagler) [inline sirepo/job_api.py]:
```py
with simulation_db.tmp_dir() as d:
     r = _request(data=(), tmpDir=d)
     if r.state != 'ok':
         return r
     f = werkzeug.secure_filename(r.file)
     return sirepo.http_reply.gen_file(d.join(f))
```

File: sirepo/simulation_db.py [hunk 5]
```diff
@@ -535,6 +535,25 @@ def process_simulation_list(res, path, data):
     ))
 
 
+def read_data_file(filepath):
```
Comment (robnagler) [inline sirepo/simulation_db.py]: This doesn't belong here. and use `mimetypes.guess_type(f.basename) or application/octect`

File: sirepo/job.py (hunk 1 above)
Comment (e-carlin) [reply sirepo/job.py]: This was a duplicate definition. See line 30.

File: sirepo/simulation_db.py (hunk 5 above)
Comment (e-carlin) [reply sirepo/simulation_db.py]: Ok, where should I put it? In job_api?

Comment (robnagler) [reply sirepo/simulation_db.py]: Yes, job_api. It's very specific.

I think we need to rename JOB_FILE (for sim.lib_file) and FILE to LIB_FILE_URI and DATA_FILE_URI. Everything associated with DATA_FILE should be data_file, DataFile, DATA_FILE, or dataFile, not just FIleReq or file. There are two different paths for different problems so they need to be distinguished clearly.

Comment (e-carlin) [reply sirepo/simulation_db.py]: Agreed. Renamed to DATA_FILE

**Comment by robnagler** [comment]
Job download data file #2050

**Pull request merged**